### PR TITLE
Update Conditions of Employment

### LINKED ISSUE #2477: Conditions of Work for MVP
The MVP requires DCMO to input candidate information for the people in their pools. For them to be able to do this, we have to match the exact data that they currently collect.

The conditions of employment list that we are currently using for MVP is instead the ideal list that they are hoping to move to in the future, but not what they have right now. So we have to revert to the list they are using now. When we release profiles this will be revamped.

The conditions of employment show up on the search page and a couple other places:

English options (for now) should be:
- Shift work
- Work weekends
- Work scheduled overtime
- Work overtime on a short notice

French options should be:
- Travailler des quarts de travail
- Travailler les fins de semaine
- Travailler des heures supplémentaires prévues
- Travailler des heures supplémentaires à court préavis

### PR DESCRIPTION
**Pull request by tristan-orourke**
Resolves #2477

This PR removes a value from the OperationalRequirements enums. If that value remains in the database anywhere, it can cause errors in GraphQL queries. To test:
1. Edit the database to enter a value of `["OVERTIME", "SHIFT_WORK"]` in the following tables & columns:
    - users.accepted_operational_requirements
    - pool_candidates.accepted_operational_requirements
    - pools.operational_requirements
    - pool_candidate_filters.operational_requirements
2. Go to http://localhost:8000/graphql-playground, run the following query. (Ensure you either have an admin bearer token, or have set AUTH_DEFAULT_USER=admin@test.com in api/.env) 
   You should get an error, something like `Expected a value of type \"OperationalRequirement\" but received: OVERTIME`
```
query {
  poolCandidates {
    acceptedOperationalRequirements
  }
  users {
    acceptedOperationalRequirements
  }
  pools {
    operationalRequirements
  }
  poolCandidateFilters {
    operationalRequirements
  }
}
```  
4. Run migrations (run `docker-compose exec -w /var/www/html/api php sh -c "php artisan migrate"` from /infrastructure folder). 
5. Running the same query from graphql-playground should now work! Viewing the database directly, you should see previous instances of "OVERTIME" replaced with "OVERTIME_SCHEDULED".

### CODE REVIEW

File: api/database/migrations/2022_04_13_190734_convert_overtime_enum.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
+<?php
+
+use App\Models\Pool;
+use App\Models\PoolCandidate;
+use App\Models\PoolCandidateFilter;
+use App\Models\User;
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
```
Comment (petertgiles) [inline api/database/migrations/2022_04_13_190734_convert_overtime_enum.php:8]: Unused imports?

File: api/database/migrations/2022_04_13_190734_convert_overtime_enum.php [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+use Illuminate\Support\Facades\Schema;
+
+class ConvertOvertimeEnum extends Migration
+{
+
+    protected function convertOperationalEnumValue($oldValue, $newValue)
+    {
+        $converEnumValue = function($item) use ($oldValue, $newValue) {
```
Comment (petertgiles) [inline api/database/migrations/2022_04_13_190734_convert_overtime_enum.php:16]: should be $convert with a t

**Review by petertgiles — APPROVED** [review]
Seems good to me!  I like the clever convert function in the migration.  A few tidyup comments but overall looks great!

**Comment by petertgiles** [comment]
I recommend merging main and review this, though.  https://github.com/GCTC-NTGC/gc-digital-talent/pull/2449/files#diff-fb251ec51456fdc6fbe09424d4f94b19ecb7dceac37971e853209b885bac2147

**Comment by tristan-orourke** [comment]
Thanks @petertgiles! I made sure to update the new OperationalRequirments in localizedConstants after the merge.

**Pull request merged**